### PR TITLE
Add LinkedData to DotcomponentsDataModel for consumption by dotcom-rendering

### PR DESCRIPTION
**Pull request by ademuk**
## What does this change?
* Add `NewsArticle`, which is similar to the existing `model.meta.WebPage`, but more specific to article pages.
* Add instance of `NewsArticle` under the new field `config.page.structuredData` in `DotcomponentsDataModel`.

## Screenshots
screen shot 2019-01-22 at 11 08 22

## What is the value of this and can you measure success?
* To allow `dotcom-rendering` to render lined data on its article pages.
* The requirement came out of a [comment on this PR](https://github.com/guardian/dotcom-rendering/pull/378#discussion_r248969255)

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Accessibility test checklist

- [x] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [x] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [x] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Comment by SiAdcock** [comment]
This looks good to me, but I'd feel better if someone with bigger Scala chops than me could also take a look 😄 

@AWare @nicl @philmcmahon ?

**Comment by ademuk** [comment]
@AWare @nicl @philmcmahon

Yes it'd be great to get some eyes on this. We've got json4s and Play Json in the mix, which has led to a bit of an inconsistency in how we serialise case classes in `LinkedData.scala`. Any pointers would be appreciated, I'm reaching the boundaries of my Scala abilities!

File: common/app/model/meta/LinkedData.scala [hunk 1]
```diff
@@ -25,7 +28,12 @@ case class Guardian(
     "https://twitter.com/guardian",
     "https://www.youtube.com/user/TheGuardian"
   )
-) extends LinkedData("Organization")
+) extends LinkedData(`@type`, `@context`)
+
+object Guardian {
+  def apply(): Guardian = new Guardian("Organization", "http://schema.org")
```
Comment (philmcmahon) [inline common/app/model/meta/LinkedData.scala]: you don't need this `apply` -case classes have an apply method by default

File: article/app/model/dotcomponents/DotcomponentsDataModel.scala [hunk 2]
```diff
@@ -53,6 +54,29 @@ case class ReaderRevenueLinks(
...
+  potentialAction: PotentialAction,
+  publisher: Guardian = Guardian(),
+  isAccessibleForFree: Boolean = true,
+) extends LinkedData(`@type`, `@context`)
+
+object NewsArticle {
+  def apply(
+   `@id`: String,
+   potentialAction: PotentialAction
+ ): NewsArticle = new NewsArticle(
```
Comment (philmcmahon) [inline article/app/model/dotcomponents/DotcomponentsDataModel.scala]: no need for `new` here

**Comment by philmcmahon** [comment]
A few scala syntax things but otherwise looks good!

**Comment by ademuk** [comment]
Thanks @philmcmahon, your comments are now addressed in my latest commit

**Review by philmcmahon — APPROVED** [review]
Worth testing on CODE briefly, but otherwise LGTM 👍

**Comment by ademuk** [comment]
@philmcmahon Looks good on CODE 👍

**Pull request merged**